Gradle build cannot be used to fully self-host Umple

The JAR that gets built by the Gradle build scripts (`quickbuild`, `fullbuild`, etc.) seems to be missing files that allow the project to be fully self-hosted. I tested this by first executing a regular build through Gradle, then manually copying the output to `libs/umple-latest.jar`. Doing this results in a sizeable number of errors (1,799 total) during the `:compileJava` task:

<details>
<summary>See build log</summary>

```sh
# With remote JAR
$ gradle quickbuild
# (output omitted; build worked just fine)

# With local JAR
$ cp dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar libs/umple-latest.jar
$ gradle quickbuild

> Task :compileTXL
txl -c /home/adam/Git/umple/UmpleToPython/txl/umpleJavaToPython.txl

> Task :downloadUmpleJar UP-TO-DATE
Download https://cruise.umple.org/umpleonline/scripts/umple.jar
Processing -> /home/adam/Git/umple/UmpleToJava/UmpleTLTemplates/Master.ump
  Finished generating Java
Success! Processed /home/adam/Git/umple/UmpleToJava/UmpleTLTemplates/Master.ump.

> Task :UmpleToJava:compileUmpleSelf
java -jar /home/adam/Git/umple/dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar /home/adam/Git/umple/UmpleToJava/UmpleTLTemplates/Master.ump

> Task :UmpleToJava:copyUmpleOutput
Copying from /home/adam/Git/umple/UmpleToJava/src-gen-UmpleTL to /home/adam/Git/umple/dist/gradle/src-gen/UmpleToJava/
Processing -> /home/adam/Git/umple/UmpleToPhp/UmpleTLTemplates/Master.ump
  Finished generating Java
Success! Processed /home/adam/Git/umple/UmpleToPhp/UmpleTLTemplates/Master.ump.

> Task :UmpleToPhp:compileUmpleSelf
java -jar /home/adam/Git/umple/dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar /home/adam/Git/umple/UmpleToPhp/UmpleTLTemplates/Master.ump
Processing -> /home/adam/Git/umple/UmpleToRTCpp/UmpleTLTemplates/Master.ump

> Task :UmpleToRTCpp:compileUmpleSelf
java -jar /home/adam/Git/umple/dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar /home/adam/Git/umple/UmpleToRTCpp/UmpleTLTemplates/Master.ump

Warning 170 on line 150 of file 'Attributes.ump':
  Finished generating Java
Custom constructors should normally not be provided in Umple. Consider using before or after keywords to adjust what the constructor does, or adjusting constructor parameters using keywords like lazy.
Success! Processed /home/adam/Git/umple/UmpleToRTCpp/UmpleTLTemplates/Master.ump.
Processing -> /home/adam/Git/umple/UmpleToRuby/UmpleTLTemplates/Master.ump
  Finished generating Java
Success! Processed /home/adam/Git/umple/UmpleToRuby/UmpleTLTemplates/Master.ump.

> Task :UmpleToRuby:compileUmpleSelf
java -jar /home/adam/Git/umple/dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar /home/adam/Git/umple/UmpleToRuby/UmpleTLTemplates/Master.ump
Processing -> /home/adam/Git/umple/UmpleToSql/UmpleTLTemplates/Master.ump
  Finished generating Java
Success! Processed /home/adam/Git/umple/UmpleToSql/UmpleTLTemplates/Master.ump.

> Task :UmpleToSql:compileUmpleSelf
java -jar /home/adam/Git/umple/dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar /home/adam/Git/umple/UmpleToSql/UmpleTLTemplates/Master.ump
Processing -> /home/adam/Git/umple/UmpleToTest/UmpleTLTemplates/Master.ump
  Finished generating Java
Success! Processed /home/adam/Git/umple/UmpleToTest/UmpleTLTemplates/Master.ump.

> Task :UmpleToTest:compileUmpleSelf
java -jar /home/adam/Git/umple/dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar /home/adam/Git/umple/UmpleToTest/UmpleTLTemplates/Master.ump
Processing -> /home/adam/Git/umple/UmpleParser/src/Master.ump
  Finished generating Java
Warning 1510 on line 347 of file 'GrammarParsing.ump':
File 'lib:GrammarParsing_Code.ump' referred to in use statement was not found
Success! Processed /home/adam/Git/umple/UmpleParser/src/Master.ump.

> Task :UmpleParser:compileUmple
java -jar /home/adam/Git/umple/libs/umple-latest.jar /home/adam/Git/umple/UmpleParser/src/Master.ump

> Task :UmpleParser:copyUmpleOutput
Copying from /home/adam/Git/umple/UmpleParser/src-gen-umple to /home/adam/Git/umple/dist/gradle/src-gen/UmpleParser/
Processing -> /home/adam/Git/umple/cruise.umple/src/Master.ump

> Task :cruise.umple:compileUmpleSelf
java -jar /home/adam/Git/umple/dist/gradle/libs/umple-1.35.0.8065.3540dc9ec.jar /home/adam/Git/umple/cruise.umple/src/Master.ump

Warning 1013 on line 132 of file 'Umple.ump':
  Finished generating Java
Parameter specification does not apply to code injections with the 'generated' keyword. The injection was applied to all generated methods.
Warning 1513 on line 117 of file 'Generator.ump':
The mixset 'RubyGeneratorIntMixset' has a use statement at line '117', however there is no declaration for this mixset. To avoid this warning, you may remove the use statement or you may add a declaration for mixset 'RubyGeneratorIntMixset'.
  Main method found. Run java DocumenterMain
  Main method found. Run java UmpleConsoleMain
  Main method found. Run java UmpleRunMain
  Main method found. Run java PlaygroundMain
  Main method found. Run java ArgumentTokenizer
  Main method found. Run java StatsMain
Success! Processed /home/adam/Git/umple/cruise.umple/src/Master.ump.

> Task :cruise.umple:copyUmpleOutput
Copying from /home/adam/Git/umple/cruise.umple/src-gen-umple to /home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/

> Task :downloadJOptSimpleVendorZip
Download https://codeload.github.com/pholser/jopt-simple/zip/jopt-simple-4.4

> Task :compileJava
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/UmpleModel.java:1895: error: cannot find symbol
      result = parser.parse(
                     ^
  symbol:   method parse(File,File[])
  location: variable parser of type RuleBasedParser
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/UmpleModel.java:1901: error: cannot find symbol
      result = parser.parse("classDefinition",""); // Fake to ensure can proceed
                     ^
  symbol:   method parse(String,String)
  location: variable parser of type RuleBasedParser
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/UmpleModel.java:1903: error: constructor ErrorMessage in class ErrorMessage cannot be applied to given types;
      result.addErrorMessage(new ErrorMessage(9000,new Position("",0,0,0),
                             ^
  required: ErrorType,Position
  found:    int,Position,String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/UmpleModel.java:1910: error: cannot find symbol
      result = parser.parse("classDefinition",""); // Fake to ensure can proceed
                     ^
  symbol:   method parse(String,String)
  location: variable parser of type RuleBasedParser
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/UmpleModel.java:1912: error: constructor ErrorMessage in class ErrorMessage cannot be applied to given types;
      result.addErrorMessage(new ErrorMessage(9000,new Position("",0,0,0),
                             ^
  required: ErrorType,Position
  found:    int,Position,String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/UmpleModel.java:2026: error: cannot find symbol
    setAnalyzers(parser.getAnalyzer().getAnalyzerMap());
                       ^
  symbol:   method getAnalyzer()
  location: variable parser of type RuleBasedParser
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:398: error: cannot find symbol
                if(subToken.is("templateExpressionContent"))
                           ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:410: error: cannot find symbol
                else if(subToken.is("templateExpression"))
                                ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:425: error: cannot find symbol
                if(token.is(TEMPLATE_EXPRESSION))
                        ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:431: error: cannot find symbol
                else if(token.is(TEMPLATE_CODEBLOCK))
                             ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:441: error: cannot find symbol
                    if(subToken.is("templateLanguageCode"))
                               ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:445: error: cannot find symbol
                    else if(subToken.is("templateExpression"))
                                    ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:454: error: cannot find symbol
                else if(token.is(TEMPLATE_INCLUDE))
                             ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:456: error: incompatible types: String cannot be converted to int
                        Token templateRefToken = token.getSubToken(TEMPLATE_INCLUDE).getSubToken("templateName");
                                                                   ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:460: error: method getValue in class Token cannot be applied to given types;
                                String refClassname = templateRefToken.getValue("classname");
                                                                      ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:464: error: method getValue in class Token cannot be applied to given types;
                                String refKey = templateRefToken.getValue("name");
                                                                ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:476: error: constructor ErrorMessage in class ErrorMessage cannot be applied to given types;
                                        parser.getParseResult().addErrorMessage(new ErrorMessage(3503,templateRefToken.getPosition(),templateName));
                                                                                ^
  required: ErrorType,Position
  found:    int,Position,String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:484: error: cannot find symbol
                else if(token.is(TEMPLATE_COMMENT))
                             ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:486: error: incompatible types: String cannot be converted to int
                  lastElement= new CommentElement(token.getPosition(), token.getSubToken(TEMPLATE_COMMENT).getValue("templateCommentContent"));
                                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:490: error: cannot find symbol
                else if(token.is(TEMPLATE_TEXT))
                             ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/TemplateAttributeParser.java:492: error: incompatible types: String cannot be converted to int
                  String text = token.getSubToken(TEMPLATE_TEXT).getValue("templateTextContent");
                                                  ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:216: error: constructor ErrorMessage in class ErrorMessage cannot be applied to given types;
          getParser().getParseResult().addErrorMessage(new ErrorMessage(errorCode,position,messages));
                                                       ^
  required: ErrorType,Position
  found:    int,Position,String[]
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:271: error: cannot find symbol
                           if(portParamToken.is("fromPort")) {
                                            ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:273: error: cannot find symbol
                           } else if(portParamToken.is("toPort")) {
                                                   ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:275: error: cannot find symbol
                           } else if(portParamToken.is("bindinHandler")) {
                                                   ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:276: error: incompatible types: String cannot be converted to int
                                   Token codeToken = portParamToken.getSubToken("code");
                                                                                ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:280: error: cannot find symbol
                           } else if(portParamToken.is("fromClassname")) {
                                                   ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:282: error: cannot find symbol
                           } else if(portParamToken.is("toClassname")) {
                                                   ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:299: error: cannot find symbol
                           if(portParamToken.is("port")) {
                                            ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:301: error: cannot find symbol
                           } else if(portParamToken.is("portDeclaration")) {
                                                   ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:317: error: cannot find symbol
                if(portParamToken.is("typedPortName")) {
                                 ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:318: error: incompatible types: String cannot be converted to int
                        Token portNameToken = portParamToken.getSubToken("portName");
                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:319: error: incompatible types: String cannot be converted to int
                        Token portTypeToken = portParamToken.getSubToken("type");
                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:320: error: incompatible types: String cannot be converted to int
                        Token portMulToken = portParamToken.getSubToken("portMultiplicity");
                                                                        ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:322: error: method getValue in class Token cannot be applied to given types;
                                name = portNameToken.getValue("portName");
                                                    ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:323: error: cannot find symbol
                                if(!Token.isValidIdentifier(name)){
                                         ^
  symbol:   method isValidIdentifier(String)
  location: class Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:324: error: method getValue in class Token cannot be applied to given types;
                                        logError(token.getPosition(), 3601, portParamToken.getValue("portName"));
                                                                                          ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:328: error: method getValue in class Token cannot be applied to given types;
                                type = portTypeToken.getValue("type");
                                                    ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:342: error: cannot find symbol
                } else if (portParamToken.is("modifier")){
                                         ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:346: error: cannot find symbol
                } else if(portParamToken.is("direction")) {
                                        ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:347: error: method getValue in class Token cannot be applied to given types;
                        String dirStr = portParamToken.getValue("direction");
                                                      ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:356: error: method getValue in class Token cannot be applied to given types;
                                logError(token.getPosition(), 3602, portParamToken.getValue("direction")); // Error msg
                                                                                  ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:358: error: cannot find symbol
                } else if(portParamToken.is("inverse")) {
                                        ^
  symbol:   method is(String)
  location: variable portParamToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:371: error: incompatible types: String cannot be converted to int
           Token codeToken = activeTriggerToken.getSubToken("code");
                                                            ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:393: error: incompatible types: String cannot be converted to int
    Token clauseToken = activeTriggerToken.getSubToken("clause");
                                                       ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:394: error: incompatible types: String cannot be converted to int
           Token timerToken = activeTriggerToken.getSubToken("timer");
                                                             ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:412: error: incompatible types: String cannot be converted to int
                   if(constraintToken.getSubToken("messageConstraint") != null) {
                                                  ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:413: error: incompatible types: String cannot be converted to int
                           Token msgToken = constraintToken.getSubToken("messageConstraint");
                                                                        ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:414: error: incompatible types: String cannot be converted to int
                           Token priorityToken = msgToken.getSubToken("priorityValue");
                                                                      ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:420: error: incompatible types: String cannot be converted to int
                   } else if(constraintToken.getSubToken("timeConstraint") != null) {
                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:421: error: incompatible types: String cannot be converted to int
                           Token timeToken = constraintToken.getSubToken("timeConstraint");
                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:422: error: incompatible types: String cannot be converted to int
                           Token clauseToken = timeToken.getSubToken("clause");
                                                                     ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:423: error: incompatible types: String cannot be converted to int
                           Token timerToken = timeToken.getSubToken("timer");
                                                                    ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:454: error: cannot find symbol
                   if(constraintToken.is("timeConstraint")) {
                                     ^
  symbol:   method is(String)
  location: variable constraintToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:455: error: incompatible types: String cannot be converted to int
                           Token timeToken = constraintToken.getSubToken("timeConstraint");
                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:456: error: incompatible types: String cannot be converted to int
                           Token clauseToken = timeToken.getSubToken("clause");
                                                                     ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:457: error: incompatible types: String cannot be converted to int
                           Token timerToken = timeToken.getSubToken("timer");
                                                                    ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:481: error: incompatible types: String cannot be converted to int
                                   Token portToken = constraintToken.getSubToken("constraintName");
                                                                                 ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:483: error: incompatible types: String cannot be converted to int
                                           Token nameToken = portToken.getSubToken("name");
                                                                                   ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:502: error: cannot find symbol
                   if (invokeToken.is("name"))
                                  ^
  symbol:   method is(String)
  location: variable invokeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:506: error: cannot find symbol
                   if (invokeToken.is("parameter"))
                                  ^
  symbol:   method is(String)
  location: variable invokeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:521: error: cannot find symbol
                   if(activeToken.is("anonymousTriggerBody")) {
                                 ^
  symbol:   method is(String)
  location: variable activeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:523: error: cannot find symbol
                   } else if(activeToken.is("hitchConstraint")) {
                                        ^
  symbol:   method is(String)
  location: variable activeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:525: error: cannot find symbol
                   } else if(activeToken.is("ConstraintTree") || activeToken.is("constraintList")) {
                                        ^
  symbol:   method is(String)
  location: variable activeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:525: error: cannot find symbol
                   } else if(activeToken.is("ConstraintTree") || activeToken.is("constraintList")) {
                                                                            ^
  symbol:   method is(String)
  location: variable activeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:527: error: cannot find symbol
                   } else if(activeToken.is("invoke")) {
                                        ^
  symbol:   method is(String)
  location: variable activeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:529: error: cannot find symbol
                   } else if(activeToken.is("thenDefinition")) {
                                        ^
  symbol:   method is(String)
  location: variable activeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:530: error: incompatible types: String cannot be converted to int
                           Token annoyToken = activeToken.getSubToken("anonymousTriggerBody");
                                                                      ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:534: error: cannot find symbol
                   } else if(activeToken.is("resolveDefinition")) {
                                        ^
  symbol:   method is(String)
  location: variable activeToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:535: error: incompatible types: String cannot be converted to int
                           Token annoyToken = activeToken.getSubToken("anonymousTriggerBody");
                                                                      ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:558: error: cannot find symbol
                  if(methodToken.is("activeDirectionHandler")) {
                                ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:559: error: incompatible types: String cannot be converted to int
                          Token methodDirectionToken = methodToken.getSubToken("direction");
                                                                               ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:571: error: cannot find symbol
                  else if (methodToken.is("inlineComment"))
                                      ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:581: error: cannot find symbol
                  else if (methodToken.is("multilineComment"))
                                      ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:593: error: cannot find symbol
                  else if (methodToken.is("reqImplementation")){
                                      ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:605: error: cannot find symbol
                  else if (methodToken.is("code"))
                                      ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:611: error: cannot find symbol
                  } else if (methodToken.is("activeTrigger")){
                                        ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:622: error: cannot find symbol
                  if (methodToken.is("methodName"))
                                 ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:626: error: cannot find symbol
                  if (methodToken.is("parameterList"))
                                 ^
  symbol:   method is(String)
  location: variable methodToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:631: error: cannot find symbol
                                  if (parameterToken.is("parameter"))
                                                    ^
  symbol:   method is(String)
  location: variable parameterToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:634: error: incompatible types: String cannot be converted to int
                                          if (parameterToken.getSubToken("type") != null)
                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:636: error: incompatible types: String cannot be converted to int
                                                  paramType = parameterToken.getSubToken("type").getValue();
                                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:638: error: incompatible types: String cannot be converted to int
                                          if (parameterToken.getSubToken("list") != null)
                                                                         ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:640: error: incompatible types: String cannot be converted to int
                                                  isList = parameterToken.getSubToken("list").getValue() != null;
                                                                                      ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:642: error: incompatible types: String cannot be converted to int
                                          String paramName = parameterToken.getSubToken("name").getValue();
                                                                                        ^
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:670: error: cannot find symbol
                if (token.is("modifier"))
                         ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:675: error: cannot find symbol
                if (token.is("ConstraintTree") || token.is("constraintList"))
                         ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:675: error: cannot find symbol
                if (token.is("ConstraintTree") || token.is("constraintList"))
                                                       ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:681: error: cannot find symbol
                else if (token.is("type"))
                              ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:686: error: cannot find symbol
                else if (token.is("activeType"))
                              ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:688: error: method getValue in class Token cannot be applied to given types;
                        String activeType = token.getValue("activeType");
                                                 ^
  required: no arguments
  found:    String
  reason: actual and formal argument lists differ in length
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:703: error: cannot find symbol
                else if (token.is("activeMethodDeclarator"))
                              ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:707: error: cannot find symbol
                else if (token.is("activeMethodBody"))
                              ^
  symbol:   method is(String)
  location: variable token of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:710: error: cannot find symbol
                        if( len > 0 && token.getSubToken(0).is("activeMethodBodyContent")) {
                                                           ^
  symbol:   method is(String)
  location: class Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:713: error: cannot find symbol
                        if( len > 1 && token.getSubToken(1).is("inverseDirectionHandler")) {
                                                           ^
  symbol:   method is(String)
  location: class Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:715: error: cannot find symbol
                                        if (subToken.is("ConstraintTree") || subToken.is("constraintList")) {
                                                    ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:715: error: cannot find symbol
                                        if (subToken.is("ConstraintTree") || subToken.is("constraintList")) {
                                                                                     ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:719: error: cannot find symbol
                                        } else if (subToken.is("activeMethodDeclarator")) {
                                                           ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:721: error: cannot find symbol
                                        } else if (subToken.is("activeMethodBodyContent")) {
                                                           ^
  symbol:   method is(String)
  location: variable subToken of type Token
/home/adam/Git/umple/dist/gradle/src-gen/cruise.umple/cruise/umple/compiler/CompositeStructureTokenAnalyzer.java:1092: error: cannot find symbol
                   if(sub.is("constraintParameterList"))
                         ^
  symbol:   method is(String)
  location: variable sub of type Token
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
100 errors
only showing the first 100 errors, of 1799 total; use -Xmaxerrs if you would like to see more

> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 36s
27 actionable tasks: 17 executed, 10 up-to-date
```

</details>

If I do the same thing but instead manually copy the _Ant_ artifact to `libs/umple-latest.jar`, then the Gradle build passes just file.

This seems to be because the Gradle JAR is missing a handful of files that are included in both the Ant JAR and the remote JAR (which presumably is the output of the Ant build).

```sh
# Remote JAR
$ jar -tf libs/umple-latest.jar | wc -l
919

# Ant JAR
$ jar -tf dist/umple.jar | wc -l
919

# Gradle JAR
$ jar -tf dist/gradle/libs/umple.jar | wc -l
847
```

Here is a list of files/directories missing from the Gradle JAR:

<details>
<summary>See file list</summary>

```sh
$ comm -23 <(jar -tf libs/umple-latest.jar | sort) <(jar -tf dist/gradle/libs/umple.jar | sort)
cruise/umple/builder/TestClass.class
cruise/umple/compiler/featureModel/
cruise/umple/compiler/mixset/
cruise/umple/implementation/
cruise/umple/implementation/alloy/
cruise/umple/implementation/AssociationSortedWithNameSpace.class
cruise/umple/implementation/ecore/
cruise/umple/implementation/gv/
cruise/umple/implementation/java/
cruise/umple/implementation/java/innerClass/
cruise/umple/implementation/java/strictness/
cruise/umple/implementation/json/
cruise/umple/implementation/nusmv/
cruise/umple/implementation/nusmvoptimizer/
cruise/umple/implementation/OneToManyUnidirectionalTest$1.class
cruise/umple/implementation/OneToManyUnidirectionalTest$Mentor.class
cruise/umple/implementation/OneToManyUnidirectionalTest$Student.class
cruise/umple/implementation/papyrus/
cruise/umple/implementation/php/
cruise/umple/implementation/php/PhpAssociationSortedWithNameSpace.class
cruise/umple/implementation/py/
cruise/umple/implementation/py/innerClassPy/
cruise/umple/implementation/py/PythonAssociationSortedWithNameSpace.class
cruise/umple/implementation/requirements/
cruise/umple/implementation/ruby/
cruise/umple/implementation/scxml/
cruise/umple/implementation/simulate/
cruise/umple/implementation/sql/
cruise/umple/implementation/test/
cruise/umple/implementation/textuml/
cruise/umple/implementation/umple/
cruise/umple/implementation/umpleself/
cruise/umple/implementation/umpleTL/
cruise/umple/implementation/use/
cruise/umple/implementation/xmi/
cruise/umple/implementation/yuml/
cruise/umple/statemachine/
cruise/umple/statemachine/implementation/
cruise/umple/statemachine/implementation/eventsequence/
cruise/umple/statemachine/implementation/java/
cruise/umple/statemachine/implementation/php/
cruise/umple/statemachine/implementation/py/
cruise/umple/statemachine/implementation/statetable/
cruise/umple/test/
cruise/umple/test/harness/
cruise/umple/test/harness/compiler/
cruise/umple/test/harness/compiler/interfaces/
cruise/umple/test/harness/compiler/interfaces/ICompiler.class
cruise/umple/test/harness/compiler/interfaces/IMemoryLanguageObjectBuilder.class
cruise/umple/test/harness/compiler/JavaInternalCompiler$1.class
cruise/umple/test/harness/compiler/JavaInternalCompiler$ClassFileManager.class
cruise/umple/test/harness/compiler/JavaInternalCompiler$CompilationErrorListener.class
cruise/umple/test/harness/compiler/JavaInternalCompiler.class
cruise/umple/test/harness/compiler/JavaMemoryObjectBuilder$InternalJavaClassMemoryObject.class
cruise/umple/test/harness/compiler/JavaMemoryObjectBuilder$JavaMemoryObjectLoader.class
cruise/umple/test/harness/compiler/JavaMemoryObjectBuilder.class
cruise/umple/test/harness/resource/
cruise/umple/test/harness/resource/TemplateGeneratedOutput.class
cruise/umple/test/harness/resource/TestParseValidation.class
cruise/umple/test/harness/resource/TestResource.class
cruise/umple/tracer/
cruise/umple/tracer/implementation/
cruise/umple/tracer/implementation/java/
cruise/umple/tracer/implementation/java/log4j/
cruise/umple/tracer/implementation/java/lttng/
cruise/umple/tracer/implementation/java/util/
cruise/umple/tracer/implementation/java/util/logging/
cruise/umple/tracer/implementation/php/
cruise/umple/util/AssertHelper.class
FileUtils.ump
GrammarParsing_Code.ump
GrammarParsing.ump
ParseUtilities_Code.ump
ParseUtilities.ump
ParsingRules_Code.ump
ParsingRules.ump
TextParser_Code.ump
TextParser.ump
```

</details>

--- a/build.gradle
+++ b/build.gradle
@@ -259,6 +259,16 @@ subprojects{
     }
     compileUmpleSelf.dependsOn(":downloadUmpleJar")
 
+    // Compile task using the remote stable jar only, never the local built jar.
+    // Used by resetUmpleSelf to break the bootstrap cycle where a locally-built
+    // incomplete jar would generate incomplete src-gen-umple output.
+    task ('compileUmpleRemote') {
+        doLast {
+            runUmpleJar(rootProject.ext.umpleJarRemoteFile, masterFile)
+        }
+    }
+    compileUmpleRemote.dependsOn(":downloadUmpleJar")
+
     // Copy task for parent project
     def fromPath = "${projectDir.toString()}/${project.ext.umpleOutputDir}"
     def intoPath = project.hasProperty("generationDir") ?
@@ -279,6 +289,7 @@ subprojects{
     }
     compileUmple.finalizedBy copyUmpleOutput
     compileUmpleSelf.finalizedBy copyUmpleOutput
+    compileUmpleRemote.finalizedBy copyUmpleOutput
 
     task copyUmpleTest (type: Copy) {
         from "${projectDir.toString()}/test"
@@ -337,6 +348,13 @@ task ('cleanUpTests') {
 task ('cleanUpUmple') {
  doLast{
     delete "${rootProject.projectDir.toString()}/dist/gradle/src-gen/cruise.umple"
+    // Also clear the local src-gen-umple so stale content from a prior Ant build
+    // does not survive into copyUmpleOutput and cause duplicate classes.
+    delete "${rootProject.projectDir.toString()}/cruise.umple/src-gen-umple"
+    // Remove the cached remote jar so downloadUmpleJar always re-fetches the
+    // real server jar. Prevents a manually-placed incomplete jar from being
+    // used by compileUmpleRemote.
+    delete "${rootProject.projectDir.toString()}/libs/umple-latest.jar"
     }
 }
 
@@ -454,9 +472,43 @@ task copyTxlFile(type: Copy) {
     }
 }
 
+// Parser .ump files are required at runtime by the Umple compiler.
+// Ant copies them explicitly; Gradle must do the same.
+task copyParserUmpFiles(type: Copy) {
+    group = "Custom"
+    description = "Copying parser .ump files that must be included in the umple jar"
+    into rootProject.ext.classfileOutputDir
+    from("UmpleParser/src") {
+        include 'GrammarParsing*.ump'
+        include 'ParseUtilities*.ump'
+        include 'ParsingRules*.ump'
+        include 'TextParser*.ump'
+    }
+    from("cruise.umple/src/util") {
+        include 'FileUtils.ump'
+    }
+    doFirst {
+        [
+            [dir: "UmpleParser/src",       pattern: 'GrammarParsing*.ump'],
+            [dir: "UmpleParser/src",       pattern: 'ParseUtilities*.ump'],
+            [dir: "UmpleParser/src",       pattern: 'ParsingRules*.ump'],
+            [dir: "UmpleParser/src",       pattern: 'TextParser*.ump'],
+            [dir: "cruise.umple/src/util", pattern: 'FileUtils.ump'],
+        ].each { spec ->
+            if (fileTree(spec.dir).include(spec.pattern).isEmpty()) {
+                throw new GradleException(
+                    "copyParserUmpFiles: no files matched '${spec.pattern}' in '${spec.dir}'. " +
+                    "The Umple JAR would be missing required runtime parser files."
+                )
+            }
+        }
+    }
+}
+
 tasks.getByName("compileJava").dependsOn 'copyDocs'
 tasks.getByName("compileJava").dependsOn 'copyReusableUmpLib'
 tasks.getByName("compileJava").dependsOn 'copyTxlFile'
+tasks.getByName("compileJava").dependsOn 'copyParserUmpFiles'
 tasks.getByName("compileJava").dependsOn ':UmpleParser:copyUmpleOutput'
 tasks.getByName("compileJava").dependsOn ':UmpleToJava:copyUmpleOutput'
 tasks.getByName("compileJava").dependsOn ':UmpleToPhp:copyUmpleOutput'
@@ -469,7 +521,7 @@ tasks.getByName("compileJava").dependsOn copyRTCppLibs
 tasks.getByName("compileJava").dependsOn getVendorLibs
 
 jar {
-    exclude 'META-INF/*.SF', 'META-INF/*.DSA', 'META-INF/*.RSA', 'META-INF/*.MF', "**/.git", "**/.ump",  "**/data"
+    exclude 'META-INF/*.SF', 'META-INF/*.DSA', 'META-INF/*.RSA', 'META-INF/*.MF', "**/.git", "**/.ump",  "**/data", "**/*Test.class"
     from rootProject.ext.classfileOutputDir
     // Set the base name -- the jar will be compiled to "${buildDir}/libs/${archivesBaseName}.jar"
     archivesBaseName = rootProject.ext.umpleCurrentJarBase
@@ -483,6 +535,7 @@ jar {
         println("Compiling to ${rootProject.ext.umpleCurrentJar} using " + rootProject.ext.classfileOutputDir)
     }
     dependsOn copyDocs
+    dependsOn copyParserUmpFiles
 }
 
 task createSymbolicLink {
@@ -1012,11 +1065,13 @@ task deploy {
     dependsOn deployUpdatedLib
 }
 
-// Builds self
+// Builds self using the remote stable jar so that the self-hosting step
+// always produces a complete src-gen-umple output regardless of whether
+// a locally-built (potentially incomplete) jar exists.
 task resetUmpleSelf {
     group = "Custom"
-    description = ""
-    dependsOn ':cruise.umple:compileUmpleSelf'
+    description = "Recompiles cruise.umple using the remote stable jar to ensure a complete self-hosted build"
+    dependsOn ':cruise.umple:compileUmpleRemote'
 }
 
 // Generates template code by first attempting to use
@@ -1037,7 +1092,7 @@ task codeGen {
 task umpleParser {
     group = "Custom"
     description = "Compiles the parser to Java"
-    dependsOn ":UmpleParser:compileUmple"
+    dependsOn ":UmpleParser:compileUmpleSelf"
 }
 
 task ('compileXUnitTemplates') {

